src/cbl-mariner/manifest.json

```diff
@@ -16,6 +16,19 @@
             }
           ]
         },
+        {
+          "platforms": [
+            {
+              "dockerfile": "src/cbl-mariner/2.0/helix/amd64",
```

Comment:
we already have plain mariner 2 machines, why do it in docker?

Reply:
(e.g. https://helix.dot.net/api/info/queues/mariner.2.amd64.open?includeQueueDepth=false&api-version=2019-06-17)

Reply:
do you have msquic on it @MattGal ? It feels like the docker is easer to manage for us.

Reply:
one more note that containers are super easy for developers for testing and investigations. Going through the helix-repro lab no so much.

Reply:
@wfurt currently the "regular" helix machines running mariner 2 on them don't have msquic, that's true, but the same "install a Centos 7 version of it" workaround could be easily applied there.  I thought we had tried this as part of https://github.com/dotnet/arcade/issues/10253 and determined it didn't work at runtime.

I'm not blocking this PR but I still think it's strange to do mariner testing in docker when we made specific efforts to have dedicated IaaS Mariner 2 test machines.  Running the tests as docker has the downsides of the per-machine cost of downloading the images on every machine as well as the bottlenecking effect of running in the same queue with all the other docker-ified linux scenarios.

Reply:
noted @MattGal. Let's revisit this when the linked MsQuic issue is fixed e.g. we have packages for Mariner. 
Also new version of MsQuic is coming out soon so we may see some more turn than usually.